src/System.Private.CoreLib/shared/System/Runtime/InteropServices/SafeHandle.cs

```diff
@@ -0,0 +1,249 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.ConstrainedExecution;
+using System.Threading;
+
+namespace System.Runtime.InteropServices
+{
+    // This implementation does not employ critical execution regions and thus cannot
+    // reliably guarantee handle release in the face of thread aborts.
+
+    /// <summary>Represents a wrapper class for operating system handles.</summary>
+    public abstract partial class SafeHandle : CriticalFinalizerObject, IDisposable
+    {
+        // IMPORTANT:
+        // - Do not add or rearrange fields as the EE depends on this layout,
+        //   as well as on the values of the StateBits flags.
+        // - The EE may also perform the same operations using equivalent native
+        //   code, so this managed code must not assume it is the only code
+        //   manipulating _state.
```

Comment:
(actually you can put it there and wrap it as `#if CORECLR` since the partial is not needed in CoreCLR)

Reply:
> Is it ok to depend on layout of something that doesn't have it explicitly marked as Sequential/Explicit

It is ok for the runtime to depend on it since the runtime controls the `Auto` layout algorithm. 

It would not be ok for user code to depend on it.